src/html5/parser/document.rs

```diff
@@ -695,9 +696,23 @@ impl DocumentBuilder {
     }
 
     /// Creates a new document fragment with the context as the root node
-    pub fn new_document_fragment(_doc: DocumentHandle, _context: NodeId) -> DocumentHandle {
-        // @TODO: Create a document fragment and set doc/context as the root node
-        todo!();
+    pub fn new_document_fragment(context: Node) -> DocumentHandle {
```

Comment:
The document builder can either return a `document` or a `document_fragment`. A document is just a simple document with a document root element. A fragment uses the `context` as a node and uses `html` as a root element.